src/quicr_client.cpp

```diff
@@ -227,6 +326,93 @@ QuicRClient::publishNamedObjectFragment(const quicr::Name& /* quicr_name */,
   throw std::runtime_error("UnImplemented");
 }
 
+bool
+QuicRClient::notify_pub_fragment(const messages::PublishDatagram& datagram,
+                                 const std::map<int, bytes>& frag_map)
+{
+  if ((frag_map.rbegin()->first & 0x1) != 0x1) {
+    return false; // Not complete, return false that this can NOT be deleted
+  }
+
+  bytes reassembled;
+
+  int seq_bytes = 0;
+  for (const auto& item : frag_map) {
+    if ((item.first >> 1) - seq_bytes != 0) {
+      // Gap in offsets, missing data, return false that this can NOT be deleted
+      return false;
+    }
+
+    reassembled.insert(
+      reassembled.end(), item.second.begin(), item.second.end());
+    seq_bytes += item.second.size();
+  }
+
+  for (const auto& entry : sub_delegates) {
+    quicr::bytes copy = reassembled;
+
+    if (entry.first.contains(datagram.header.name)) {
+      sub_delegates[entry.first]->onSubscribedObject(
+        datagram.header.name, 0x0, 0x0, false, std::move(copy));
+    } else {
+      std::cout << "Name:" << datagram.header.name.to_hex()
+                << ", not in namespace " << entry.first.to_hex() << std::endl;
+    }
+  }
+
+  return true;
+}
+
+void
+QuicRClient::handle_pub_fragment(messages::PublishDatagram&& datagram)
+{
+  static int cindex = 1;
+
+  // Check the current index first considering it's likely in the current buffer
+  const auto& msg_iter = fragments[cindex].find(datagram.header.name);
+  if (msg_iter != fragments[cindex].end()) {
+    // Found
+    msg_iter->second.emplace(from_varint(datagram.header.offset_and_fin),
+                             std::move(datagram.media_data));
+    if (notify_pub_fragment(datagram, msg_iter->second))
+      fragments[cindex].erase(msg_iter);
+
+  } else {
+    // Not in current buffer, search all buffers
+    bool found = false;
+    for (auto& buf : fragments) {
+      const auto& msg_iter = buf.second.find(datagram.header.name);
+      if (msg_iter != buf.second.end()) {
+        // Found
+        msg_iter->second.emplace(from_varint(datagram.header.offset_and_fin),
+                                 std::move(datagram.media_data));
+        if (notify_pub_fragment(datagram, msg_iter->second)) {
+          fragments[cindex].erase(msg_iter);
+        }
+        found = true;
+        break;
+      }
+    }
+
+    if (!found) {
+      // If not found in any buffer, then add to current buffer
+      fragments[cindex][datagram.header.name].emplace(
+        from_varint(datagram.header.offset_and_fin),
+        std::move(datagram.media_data));
+    }
+  }
+
+  // Move to next buffer if reached max
+  if ((int)fragments[cindex].size() >= MAX_FRAGMENT_NAMES_PENDING_PER_BUFFER) {
```

Comment:
Why c-style cast to int? Why isn't `MAX_FRAGMENT_NAMES_PENDING_PER_BUFFER` unsigned?

Reply:
No, its signed... although, it can be changed to unsigned. What are you thoughts on why it needs to be unsigned?

Reply:
Validation sounds good. I commented because it's generally [frowned upon to use c-style casting](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast) these days (and in fact [avoiding cast when you can](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es48-avoid-casts)). Changing `MAX_FRAGMENT_NAMES_PENDING_PER_BUFFER` to unsigned should do that. And doing some validation on input/config as you suggest is good too.

Reply:
I just changed it to unsigned as it doesn't have to be signed. Cast removed.